code/modules/power/apc/apc_main.dm

```diff
@@ -229,6 +229,7 @@
 /obj/machinery/power/apc/Destroy()
 	if(malfai && operating)
 		malfai.malf_picker.processing_time = clamp(malfai.malf_picker.processing_time - 10, 0, 1000)
+		malfai.hacked_apcs -= src
```

Comment:
```suggestion
	if(malfai)
		if(operating)
			malfai.malf_picker.processing_time = clamp(malfai.malf_picker.processing_time - 10, 0, 1000)
		malfai.hacked_apcs -= src
		malfai = null
```
We want to handle the references regardless of whether this is oeprating or not.

Also we never set malfai to `null` I think?